fields.go

```diff
@@ -88,6 +90,20 @@ func Error(err error) Field {
 	return NamedError("error", err)
 }
 
+// ptrFieldsTypes lists all acceptable pointer types for creating fields.
+//
+// Caveat: we can't use ~type because the type assertion switch in zap.Any will fail with a
+// custom type.
+type ptrFieldsTypes interface {
+	*string | *int | *int32 | *int64 | *uint | *uint32 | *uint64 | *float32 | *float64 | *bool | *time.Time | *time.Duration
```

Comment:
It's the first time I see this notation. What would it be called? Having trouble finding docs on it.

Reply:
Here you go! https://github.com/akutz/go-generics-the-hard-way/blob/main/03-getting-started/06-tilde.md

Reply:
I was referring to the pipe symbols in the interface, but what you sent was a valuable lesson regardless, so thank you 😄